react/components/Range.js

```diff
@@ -0,0 +1,297 @@
+    const { max, min } = this.props
+    const rect = this.sliderRef.current.getBoundingClientRect()
+    const percentageComplete = (value - min) / (max - min)
+
+    let translatePx = percentageComplete * rect.width
+
+    if (position === 'right') {
+      translatePx = rect.width - translatePx
+    }
+
+    return translatePx
+  }
+
+  handleDragStart = position => e => {
+    if (this.props.disabled) {
+      return
+    }
+
+    this.setState({
+      dragging: position,
+    })
+
+    this.valuesBeforeDrag_ = this.state.values
+
+    // https://reactjs.org/docs/events.html#event-pooling
+    e.persist()
+
+    const moveHandler = this.handleDrag(position)
+
+    this.cancelDragEvent_ = () => {
+      this.valuesBeforeDrag_ = undefined
+      UP_EVENTS.forEach(evtName => document.body.removeEventListener(evtName, handleUpEvent))
+      document.body.removeEventListener(MOVE_EVENT_MAP[e.type], moveHandler)
+      document.body.removeEventListener('keydown', this.handleKeyDown)
+    }
+
+    const handleUpEvent = () => {
+      this.cancelDragEvent_()
+      this.handleDragEnd()
+    }
+
+    UP_EVENTS.forEach(evtName => document.body.addEventListener(evtName, handleUpEvent))
+    document.body.addEventListener(MOVE_EVENT_MAP[e.type], moveHandler)
+    document.body.addEventListener('keydown', this.handleKeyDown)
+
+    this.updatePositionFromEvent(e, position)
+  }
+
+  updatePositionFromEvent = (e, position) => {
+    const slider = this.sliderRef.current
+    const rect = slider.getBoundingClientRect()
+
+    const xPos = getPageX(e) - rect.left
+
+    const percentageComplete = xPos / rect.width
+
+    const value = this.getValueForPercent(percentageComplete, position)
+
+    this.updatePositionForValue(value, position)
+  }
+
+  handleDrag = position => e => {
+    e.preventDefault()
+    this.updatePositionFromEvent(e, position)
+  }
+
+  updatePositionForValue = (value, position) => {
+    const translatePx = this.getTranslateValueForInputValue(value, position)
+
+    requestAnimationFrame(() => {
+      this.setState({
+        values: {
+          ...this.state.values,
+          [position]: value,
+        },
+        translate: {
+          ...this.state.translate,
+          [position]: translatePx,
+        },
+      })
+    })
+  }
+
+  handleDragEnd = () => {
+    this.setState({
+      dragging: null,
+    })
+
+    this.cancelDragEvent_ = undefined
+
+    this.props.onChange(this.state.values)
+  }
+
+  handleKeyDown = evt => {
+    if (!isEscKeyEvent(evt) || !this.state.dragging) {
+      return
+    }
+
+    this.setState({
+      dragging: false,
+      values: this.valuesBeforeDrag_,
+    })
+
+    this.cancelDragEvent_()
+    this.cancelDragEvent = undefined
+
+    this.updateLayout()
+  }
+
+  render() {
+    const { disabled } = this.props
+    const { left, right } = this.state.translate
+
+    return (
+      <div className="ph4">
+        <div className="vtex-range w-100 relative pv4" style={{ height: 48 }}>
```

Comment:
you can use h2 `height: 32px` of tachyons and set a padding higher, like pv7

Reply:
It doesn't work, I need at least 48px of height for the range to not visually break.